lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -459,6 +468,44 @@ def wait_for_draining(self, mongos_hosts: List[str]):
 
                 break
 
+    def get_shard_status(self):
+        """Returns the current status of the shard.
+
+        Note: No need to report if currently draining, since that check block other hooks from
+        executing.
+        """
+        if not self.charm.is_role(Config.Role.SHARD):
+            logger.info("skipping status check, charm is not running as a shard")
+            return None
+
+        if not self.charm.db_initialised:
+            logger.info("No status for shard to report, waiting for db to be initialised.")
+            return None
+
+        if self.model.get_relation(LEGACY_REL_NAME):
+            return BlockedStatus(f"relation {LEGACY_REL_NAME} to shard not supported.")
+
+        if self.model.get_relation(REL_NAME):
+            return BlockedStatus(f"relation {REL_NAME} to shard not supported.")
+
+        if not self.model.get_relation(self.relation_name) and not self.charm.drained:
+            return BlockedStatus("missing relation to config server")
+
+        if not self.model.get_relation(self.relation_name) and self.charm.drained:
+            return ActiveStatus("Shard drained from cluster, ready for removal")
+
+        if not self._is_mongos_reachable():
+            return BlockedStatus("Config server unreachable")
+
+        if not self._is_added_to_cluster():
+            self.charm.unit.status = MaintenanceStatus("Adding shard to config-server")
```

Comment:
is applying the status here and **not** returning  expected?

Reply:
good catch! thank you! 🙇